File: ODatav4/wwwroot/bridge/startup.js
sap.ui.getCore().attachInit(function () {
    sap.ui.require(System.Array.init(["sap/ui/core/tutorial/root/mockserver", "sap/m/Shell", "sap/ui/core/ComponentContainer"], System.String), function (mockserver) {
        // Requests to the service are intercepted and fake responses returned.
        // Comment this out to connect to the real service.
        mockserver.start();

        new sap.m.Shell({ app: new sap.ui.core.ComponentContainer({ height: "100%", component: sap.ui.component({ name: "sap.ui.core.tutorial.odatav4", manifestFirst: true, id: "odatav4" }) }) }).placeAt("content");
    });
});

Bridge.assembly("ODatav4ClientScripts", function ($asm, globals) {
    "use strict";

    Bridge.define("sap.ui.core.tutorial.odatav4.startup");
});

File: ODatav4ClientScripts/Startup.cs
using Bridge;
using System;
using System.Linq;
using UI5;

namespace ODatav4ClientScripts
{
    [Reflectable(false)]
    [FileName("startup.js")]
    [Name("sap.ui.core.tutorial.odatav4.startup")]
    public class Startup
    {
        [Init(InitPosition.Top)]
        public static void StartupScript()
        {
            sap.ui.getCore().attachInit(() => {
                sap.ui.require(new string[] {
                    "sap/ui/core/tutorial/root/mockserver",
                    "sap/m/Shell",
                    "sap/ui/core/ComponentContainer"
                }, new Action<Mockup>(
                    (Mockup mockserver) => {
                        // Requests to the service are intercepted and fake responses returned.
                        // Comment this out to connect to the real service.
                        mockserver.start();

                        new sap.m.Shell(new sap.m.Shell.Settings() {
                            app = new sap.ui.core.ComponentContainer(new sap.ui.core.ComponentContainer.Settings() {
                                height = "100%",
                                component = (sap.ui.core.UIComponent)sap.ui.component(new sap.ui.ComponentConfig() {
                                    name = "sap.ui.core.tutorial.odatav4",
                                    manifestFirst = true,
                                    id = "odatav4"
                                })
                            })
                        }).placeAt("content");
                    }
                ));
            });
        }
    }

    [External]
    [Name("sap.ui.core.tutorial.root.mockserver")]
    public class Mockup
    {
        public extern void start();
    }
}